Formula/pnpm.rb

```diff
@@ -20,14 +20,66 @@ class Pnpm < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:  "52d1cbcba6a433d4f18eef3d3efafe44ac46b41fcbbc6be1956109ce69cb305f"
   end
 
-  depends_on "node"
+  depends_on "node@14" => :build
+
+  # Described in https://github.com/pnpm/pnpm#installation
+  # Managed by https://github.com/pnpm/get
+  resource "pnpm-buildtime" do
+    url "https://get.pnpm.io/v6.7.js"
+    sha256 "56bffe269aab055bec3a07fa65c387f5a35729550dc7fc76711844a2de6e1b8b"
+  end
 
   def install
-    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
-    bin.install_symlink Dir["#{libexec}/bin/*"]
+    buildtime_bin = buildpath/"buildtime-bin"
+    resource("pnpm-buildtime").stage do |r|
+      buildtime_bin.install "v#{r.version}.js"
+      (buildtime_bin/"pnpm").write_env_script "#{Formula["node@14"].bin}/node",
+        buildpath/"buildtime-bin/v#{r.version}.js", {}
+      chmod 0755, buildtime_bin/"pnpm"
+    end
+    ENV.prepend_path "PATH", buildtime_bin
+    system "pnpm", "install"
+    chdir "packages/pnpm" do
+      system buildpath/"node_modules/.bin/tsc", "--build"
+      system "pnpm", "run", "bundle"
+      on_macos do
+        system "node_modules/.bin/pkg", "--target=host", "--out-path=dist", "dist/pnpm.cjs"
+      end
+      on_linux do
+        system "node_modules/.bin/pkg", "--target=linuxstatic", "--out-path=dist", "dist/pnpm.cjs"
+      end
+      bin.install "dist/pnpm"
+    end
+  end
+
+  def caveats
+    <<~EOS
+      You should create npm's global directory somewhere if you
+      want to install packages globally:
+
+        mkdir -p ~/.npm-packages/bin
+
+      Run the following command:
+
+        echo 'prefix=${HOME}/.npm-packages' >> ~/.npmrc
+
+      Add the following to #{shell_profile} or your desired shell
+      configuration file:
+
+        export PATH="$HOME/.npm-packages/bin:$PATH"
+
+      You can set prefix to any location, but leaving it unchanged from
+      #{HOMEBREW_PREFIX} will destroy any Homebrew-installed Node installations
+      upon upgrade/reinstall.
```

Comment:
We should set the prefix the something like a tmp directory then.

Reply:
The pnpm users would want the pnpm-managed executables persistent, so the tmp directory does not make sense.
I will reword this section.

Reply:
I get that users want to set it to something else, but the default should absolutely not be breaking homebrew installed tools.

Reply:
Indeed.

Reply:
@zkochan Can you come up with some ideas?
The situation is as follows:

- `/usr/local/bin` is writable with an ordinal user and is in PATH.
- `/usr/local/bin/node` can exists.
- With Homebrew, pnpm is not allowed to write anything under `/usr/local/bin`, though it is writable and in PATH.

Reply:
so if there is node installed with homebrew and someone runs `pnpm env use --global lts`, pnpm should not override the nodejs that was installed by homebrew?

Reply:
Really there should never be a case where running a tool overrides a package managers content.

Reply:
pnpm is a package manager as well. We could either use a different location for the binaries installed by pnpm (which is I believe a good solution). Or we could check if the specific binary was installed by pnpm. If not, don't override it.

Reply:
An alternative location is good since I don't think this will work well on other systems either. In most cases this directory will be owned by root.